components/core/src/clp_s/clp-s.cpp

```
@@ -129,6 +157,162 @@
+
+template <typename T>
+auto run_serializer(clp_s::JsonToIrParserOption const& option, std::string path) {
+    auto result{Serializer<T>::create()};
+    if (result.has_error()) {
+        SPDLOG_ERROR("Failed to create Serializer");
+        return false;
+    }
+    auto& serializer{result.value()};
+    std::ifstream in_file;
+    in_file.open(path, std::ifstream::in);
+    if (false == in_file.is_open()) {
```

cppcheck (2.10-2)

[error] Exception thrown in function declared not to throw exceptions.

(throwInNoexceptFunction)